framework/src/actions/MaterialOutputAction.C

```diff
@@ -57,28 +59,22 @@ MaterialOutputAction::MaterialOutputAction(InputParameters params)
 void
 MaterialOutputAction::act()
 {
-  // Error if _problem is NULL, I don't know how this would happen
-  if (_problem.get() == NULL)
-    mooseError("FEProblemBase pointer is NULL, it is needed for auto material property output");
-
-  buildMaterialOutputObjects(_problem.get());
-}
+  mooseAssert(_problem,
+              "FEProblemBase pointer is nullptr, it is needed for auto material property output");
 
-void
-MaterialOutputAction::buildMaterialOutputObjects(FEProblemBase * problem_ptr)
-{
   // Do nothing if the application does not have output
```

Comment:
This should get removed, it is possible to have Output objects without "add_output" action.

Reply:
right - removed!